src/main/java/de/tum/in/www1/artemis/service/ResourceLoaderService.java

```diff
@@ -136,17 +137,30 @@ public Resource[] getResources(final Path basePath, final String pattern) {
         return Objects.requireNonNullElseGet(resources, () -> new Resource[0]);
     }
 
+    private Resource[] getFileResources(final String resourceLocation) throws IOException {
+        final var fileAndDirectoryResources = resourceLoader.getResources(resourceLocation);
+
+        return Arrays.stream(fileAndDirectoryResources).filter(r -> {
+            try {
+                return r.getFile().isFile();
+            }
+            catch (IOException e) {
```

Comment:
```suggestion
            catch (IOException e) {
                log.warn("Could not load resource {}.", r, e);
```
Should we log a warning here? I’m not really sure when such an exception could occur when loading data from the JAR, though.

If logging: not sure if `r` already has a nice `toString` implementation that shows the name, or if it needs a `r.getFile()`/`r.getPath()` or whatever suitable methods it has.

Reply:
I was just trying to satisfy the Streaming-API here. This exception really shouldn't occur.

Thinking more about it, this isn't the right place to handle such an exception. Since the method is already throwing `IOException` we can ignore it here and handle it further up. Implemented in [4130317](https://github.com/ls1intum/Artemis/commit/413031710303c60a761fc0567852084bbdea719b).